core-bundle/src/Resources/contao/modules/ModuleTwoFactor.php

```diff
@@ -63,12 +69,34 @@ protected function compile()
 			$this->disableTwoFactor($user, $return);
 		}
 
+		if (Input::post('FORM_SUBMIT') == 'tl_two_factor_show_backup_codes' && $security->isGranted('IS_AUTHENTICATED_FULLY'))
```

Comment:
We are not checking `$security->isGranted('IS_AUTHENTICATED_FULLY')` for the two cases above (enable and disable 2FA) and we are not checking it at all in the `TwoFactorController` class. We should check it everywhere though, shouldn't we?

Reply:
Like in 69fceed05?